src/threading.ts

```diff
@@ -117,12 +118,33 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(f: Py_FrameObject): void {
```

Comment:
> But the comparison between exception and exception handler type occurs only in `COMPARE_OP` opcode. So `tryCatchException` only finds an exception handler.

That's fine; what's wrong with re-throwing? Seems like, semantically, the above is equivalent to (and I could be wrong):

```{python}
try:
    raise NameError
except:
    if sys.exc_info()[0] is ArithmeticError:
        # handle here
    else:
        raise sys.exc_info()[0]
```

The JVM does both in one go, CPython does one and then the other. Both can be represented using a simple `throwException` function that handles unwinding the stack.

And since exceptions are supposed to be a rare event, throwing it twice in Python isn't a serious performance issue.